SNR/plot_rawacf_lag_power.py

```diff
@@ -39,15 +36,60 @@ def usage_msg():
 def plot_parser():
     parser = argparse.ArgumentParser(usage=usage_msg())
     parser.add_argument("rawacf_file", help="Name of the file to plot.")
+    parser.add_argument("--beams", help="Beams to plot. Format as --beams=0,1,2-5", type=str)
+    parser.add_argument("--lags", help="Lags to plot. Format as --lags=0,1-4,11", type=str, default='0')
+    parser.add_argument("--datasets", help="Datasets to plot. Format as --datasets=main_acfs,intf_acfs,xcfs", type=str)
+    parser.add_argument("--max-power", help="Maximum Power of color scale (dB).", default=50.0, type=float)
+    parser.add_argument("--min-power", help="Minimum Power of color scale (dB).", default=10.0, type=float)
+    parser.add_argument("--plot-directory", help="Directory to save plots.", default='', type=str)
+    parser.add_argument("--figsize", help="Figure dimensions in inches. Format as --figsize=10,6", type=str)
+    parser.add_argument("--num-processes", help="Number of processes to use for plotting.", default=3, type=int)
     return parser
 
 
 if __name__ == '__main__':
-    parser = plot_parser()
-    args = parser.parse_args()
+    rawacf_parser = plot_parser()
+    args = rawacf_parser.parse_args()
 
     filename = args.rawacf_file
-    
-    plot_rawacf_lag_pwr(filename) # plot all beams
 
+    beam_nums = []
+    if args.beams is not None:
```

Comment:
these snippets of list-building code for beams, lags, antennas, etc are good candidates for making a method to build a list out of a string like this: '0,1,3-7,10,12'